### ISSUE TITLE
legacy_objc_type rule triggers on NSNumber in the context of NumberFormatter which requires it

### ISSUE THREAD

**Issue by tylerbarnesTrexis** [realm/SwiftLint#3723]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

legacy_objc_type rule triggers on NSNumber in the context of NumberFormatter

##### Complete output when running SwiftLint, including the stack trace and command used

```zsh
% swiftlint
Linting Swift files in current working directory
Linting 
...
Linting
warning: Legacy Objective-C Reference Type Violation: Prefer Swift value types to bridged Objective-C reference types (legacy_objc_type)
Done linting! Found 1 violation, 0 serious in files.
```

### Environment

* SwiftLint version: 0.44.0
* Installation method used: Homebrew and SPM via Custom Package
* Paste your configuration file:

```yml
only_rules:
  - legacy_objc_type
```

```swift
// This triggers a violation:
NumberFormatter.string(from: Double as NSNumber)
```

_Originally posted by @PH9 in https://github.com/realm/SwiftLint/issues/3555#issuecomment-811875997_

**Comment by CraigSiemens** [comment]
We recently ran into this as well with `NumberFormatter.string(from:)` and `UNMutableNotificationContent.badge` both still expecting a `NSNumber`.

**Comment by SimplyDanny** [comment]
In my opinion, there is nothing we can really do about that. When working with certain frameworks it is simply unavoidable to reference these types in some way. The rule is opt-in and can be disabled by comments in the code in case of (rare) exceptions.

**Comment by userException** [comment]
I got an error for NSNumber violation so changed:
NSNumber.init(value: someValue)
to
NSNumber.(value: someValue)

Worked !

**Comment by rgoldberg** [comment]
@SimplyDanny Could a `ignore` config key accept an array of type names to ignore?

If you want to ignore this rule for `NSNumber`s because your Swift code must use many `NSNumber`s, just configure:

```
legacy_objc_type:
  ignore: [NSNumber]
```

The rule will still trigger for all other bridged Objective-C types.

**Comment by SimplyDanny** [comment]
> @SimplyDanny Could an `ignore` or `exclude` config key accept an array of type names to ignore?

That's still not a fix but at least a way to reduce the number of false positives. I'd be supportive if someone wants to add such an option.

**Comment by kapitoshka438** [comment]
> > [@SimplyDanny](https://github.com/SimplyDanny) Could an `ignore` or `exclude` config key accept an array of type names to ignore?
> 
> That's still not a fix but at least a way to reduce the number of false positives. I'd be supportive if someone wants to add such an option.

@SimplyDanny please review
